handler/oauth2/flow_authorize_code_token.go

```diff
@@ -170,22 +170,20 @@ func (c *AuthorizeExplicitGrantHandler) PopulateTokenEndpointResponse(ctx contex
 	if err != nil {
 		return errorsx.WithStack(fosite.ErrServerError.WithWrap(err).WithDebug(err.Error()))
 	}
-
-	if err := c.CoreStorage.InvalidateAuthorizeCodeSession(ctx, signature); err != nil {
-		if rollBackTxnErr := storage.MaybeRollbackTx(ctx, c.CoreStorage); rollBackTxnErr != nil {
-			return errorsx.WithStack(fosite.ErrServerError.WithWrap(err).WithDebugf("error: %s; rollback error: %s", err, rollBackTxnErr))
+	defer func() {
+		if err != nil {
+			if rollBackTxnErr := storage.MaybeRollbackTx(ctx, c.CoreStorage); rollBackTxnErr != nil {
+				err = errorsx.WithStack(fosite.ErrServerError.WithWrap(err).WithDebugf("error: %s; rollback error: %s", err, rollBackTxnErr))
+			}
 		}
+	}()
```

Comment:
This wouldn't roll back on a panic, it rewires the logic from explicit rollback to implicit via defer. A panic would need a recover on the top level: https://go.dev/blog/defer-panic-and-recover

Reply:
Of course it would. From the link you posted itself (emphasis mine):

> Panic is a built-in function that stops the ordinary flow of control and begins panicking. When the function F calls panic, execution of F stops, **any deferred functions in F are executed normally**, and then F returns to its caller. To the caller, F then behaves like a call to panic. The process continues up the stack until all functions in the current goroutine have returned, at which point the program crashes.

Reply:
In German we say: "Wer lesen kann ist klar im Vorteil" (loosely translated to "Who can read has a clear advantage"). You're absolutely right, this does indeed catch any panic happening!

Reply:
:-)